### DISCUSSION TITLE
Application type

### DISCUSSION THREAD

**Discussion by Michael21107** [scripthookvdotnet/scripthookvdotnet#1357] [Q&A] [answered]
V1 is an ASI without xml reference and V2 and V3 are DLLs with XML references, correct?

**Comment by kagikn** [comment, marked as answer]
Not really, generated XML files are solely for documentation for developers (both script devs and SHVDN code devs). SHVDN uses the API dll for V2 when it finds script dolls that references to a `ScriptHookVDotNet` assembly with the version `0.0.0.0` (exact match in both name and version). SHVDN didn't have version info until v2.10.0 has its unique version info.

**Reply by Michael21107** [reply]
gotcha, so may i ask what is shvdn asi for?

**Reply by kagikn** [reply]
The asi module uses a tick event from SHV, manages all scripts, handles shared native memory stuff (such as entity pools), and operates its console. 

Well, I think I'll separate all of the stuff except for SHV tick stuff into pure .NET module so we can introduce NuGet references if needed and we can use a wrapper library for ImGui, but not right now…

**Reply by Michael21107** [reply]
gotcha, thank you
